python/ngen_cal/src/ngen/cal/calibration_set.py

```diff
@@ -25,23 +25,29 @@ class CalibrationSet(Evaluatable):
         A HY_Features based catchment with additional calibration information/functionality
     """
 
-    def __init__(self, adjustables: Sequence[Adjustable], eval_nexus: Nexus, hooks: ModelHooks, start_time: str, end_time: str, eval_params: 'EvaluationOptions'):
-        """
-
-        """
+    def __init__(self, adjustables: Sequence[Adjustable], eval_nexus: Nexus, hooks: ModelHooks, start_time: datetime, end_time: datetime, eval_params: EvaluationOptions):
         super().__init__(eval_params)
         self._eval_nexus = eval_nexus
         self._adjustables = adjustables
         # record the hooks needed for output and checkpointing
         self._hooks = hooks
 
-        #use the nwis location to get observation data
-        obs =self._eval_nexus._hydro_location.get_data(start_time, end_time)
-        #make sure data is hourly
-        self._observed = obs.set_index('value_time')['value'].resample('1h').nearest()
-        self._observed.rename('obs_flow', inplace=True)
-        #observations in ft^3/s convert to m^3/s
-        self._observed = self._observed * 0.028316847
+        # use the nwis location to get observation data
+        from hypy.hydrolocation.nwis_location import NWISLocation
+        location = self._eval_nexus._hydro_location
+        assert isinstance(location, NWISLocation), f"expected hypy.hydrolocation.NWISLocation instance, got {type(location)}"
+
+        # TODO: derive this from realization config
+        simulation_interval: pd.Timedelta = pd.Timedelta(3600, unit="s")
+        obs = self._hooks.ngen_cal_model_observations(
+            id=location.station_id,
+            start_time=start_time,
```

Comment:
In reality I think the caller should pass `start_time + 1 dt` (e.g. `3600s`) since the model outputs will not contain values for the actual `start_time` (left exclusive). I don't think this is a problem now, but just wanted to note it.

Reply:
It does matter and things will break if we don't properly account for this. This only works right now b.c. `csv_output`'s `dt` is `300s` which means the first value is `start_time` + `5min`. When we resample the simulation output to the hour using `.resample("h").first()` the simulated value at `5min` is backfilled to `start_time` and lines up with the nwis observations.

Reply:
This likely heavily depends on how the data is "merged", in search.py `_objective_function`.  Currently this done with a pandas `merge` with `left_index=True, right_index=True` which should result in a data frame of the overlapping indicies.